jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2064,7 +2064,39 @@ ecma_op_object_get_property_names (ecma_object_t *obj_p, /**< object */
       }
       else if (!prop_is_symbol && (opts & ECMA_LIST_SYMBOLS_ONLY) == 0)
       {
-        ecma_collection_push_back (return_keys, entry);
+        if ((opts & ECMA_LIST_ENUMERABLE) == 0)
+        {
+          ecma_collection_push_back (return_keys, entry);
+        }
+        else
+        {
+          ecma_property_descriptor_t prop_desc;
+          ecma_value_t status = ecma_proxy_object_get_own_property_descriptor (obj_p, prop_name_p, &prop_desc);
+
+          if (ECMA_IS_VALUE_ERROR (status))
+          {
+            ecma_free_value (status);
```

Comment:
freeing the error representing value is unnecessary.

Reply:
Done.